superset-frontend/src/dashboard/components/Header/index.jsx

```diff
@@ -394,10 +451,10 @@ class Header extends React.PureComponent {
       setRefreshFrequency,
       lastModifiedTime,
     } = this.props;
-
     const userCanEdit = dashboardInfo.dash_edit_perm;
     const userCanShare = dashboardInfo.dash_share_perm;
     const userCanSaveAs = dashboardInfo.dash_save_perm;
+    const canAccesReportsModal = !editMode && this.canAddReportsModal();
```

Comment:
I would use something like "shouldShow..." instead of "canAccess".. the latter has more of a connotation around permissions, which they could have permissions to see it, just they're in the wrong mode.

Reply:
that makes sense, I like that wording change.